### PR TITLE
adding methods to aid in token setup

### PR DESCRIPTION
**Pull request by fcollman**
I'm trying to make it easier/more intuitive for users to setup tokens and support multiple databases.  This adds a new method for setting up a token which doesn't create a new one by default, and saves a token specific to the server by default, so now users will save and load those files as they interact with multiple servers and they can utilize the setup function either way.

### CODE REVIEW

File: caveclient/auth.py [hunk 1]
```diff
@@ -21,6 +22,28 @@
...
+            secrets = json.load(f)
+
+        if overwrite is False and key in secrets:
+            raise ValueError(f'Key "{key}" already exists in token file "{filepath}"')
+    else:
+        secrets = {}
+
+    secrets[key] = token
+
+    secret_dir, _ = os.path.split(filepath)
```
Comment (sdorkenw) [inline caveclient/auth.py:38]: better to use `os.path.dirname`

File: caveclient/auth.py [hunk 2]
```diff
@@ -118,6 +141,54 @@ def _load_token(self, token_file, token_key):
...
+                1) Go to: {auth_url} to view a list of your existing tokens.
+                2) Log in with your Google credentials copy one of the tokens from the dictionary (the string under the key 'token').
+                3a) Save it to your computer with: client.auth.save_token(token="PASTE_YOUR_TOKEN_HERE")
+                or
+                3b) Set it for the current session only with client.auth.token = "PASTE_YOUR_TOKEN_HERE"
+                Note: If you need to save or load multiple tokens, please read the documentation for details.
+                if you want to create a new token, or have no token use ```self.get_new_token``` instead
+                or use this function with the keyword argument make_new=True"""
+        print(txt)
+        if open is True:
```
Comment (sdorkenw) [inline caveclient/auth.py:172]: better: `if open:`

**Review by sdorkenw — APPROVED** [review]
LGTM

File: caveclient/auth.py (hunk 1 above)
Comment (fcollman) [reply caveclient/auth.py]: fixed

File: caveclient/auth.py (hunk 2 above)
Comment (fcollman) [reply caveclient/auth.py]: fixed

**Pull request merged**